rocketpy/mathutils/function.py

```diff
@@ -488,19 +488,18 @@ def get_value_opt(x):
 
             # change the function's name to avoid mypy's error
             def get_value_opt_multiple(*args):
-                x = np.array([[float(x) for x in list(args)]])
-                numerator_sum = 0
-                denominator_sum = 0
-                for i in range(len_y_data):
-                    sub = x_data[i] - x
-                    distance = np.linalg.norm(sub)
-                    if distance == 0:
-                        numerator_sum = y_data[i]
-                        denominator_sum = 1
-                        break
-                    weight = distance ** (-3)
-                    numerator_sum = numerator_sum + y_data[i] * weight
-                    denominator_sum = denominator_sum + weight
+                x = np.array([[float(val) for val in args]])
+                sub_matrix = x_data - x
+                distances_squared = np.sum(sub_matrix**2, axis=1)
+
+                zero_distance_index = np.where(distances_squared == 0)[0]
+                if len(zero_distance_index) > 0:
```

Comment:
when does this condition doesn't get true?

Reply:
if len(zero_distance_index) > 0:

- is True when x (which comes from *args) is an element of x_data. In this case you just have to return the corresponding value of y_data.
- is False when x is not exactly an element of x_data. In this case you have to return an output based on the weighted average calculation with distances.


If the tests do not cover the 3 lines below, maybe it's because there are only tests with values directly from the source used to create the Function object.


You can try it with the example : 

```
import numpy as np


def get_value_opt_multiple(*args):
    x = np.array([[float(val) for val in args]])
    sub_matrix = x_data - x
    distances_squared = np.sum(sub_matrix**2, axis=1)

    zero_distance_index = np.where(distances_squared == 0)[0]
    if len(zero_distance_index) > 0:
        print("Input in the source")
        return y_data[zero_distance_index[0]]
    
    print("Input not in the source")
    weights = distances_squared ** (-1.5)
    numerator_sum = np.sum(y_data * weights)
    denominator_sum = np.sum(weights)

    return numerator_sum / denominator_sum


# Sample data
x_data = np.array([[1.0, 2.0], [3.0, 4.0], [5.0, 6.0]])
y_data = np.array([10.0, 20.0, 30.0])


# Input in the source
args = (3.0, 4.0)
result = get_value_opt_multiple(*args)
print("Result:", result)

# Input not in the source
args = (4.0, 5.0)
result = get_value_opt_multiple(*args)
print("Result:", result)
```


The output is : 

```
Input in the source
Result: 20.0
Input not in the source
Result: 24.727272727272723
```